### PR TITLE
Azure files - add support to run the tests for in-tree

### PR DESCRIPTION
**Pull request by kkmsft**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind test

> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Azure files - add support to run the tests for in-tree
Document steps to run azure file e2e in-tree tests

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Note: This PR is focussed on functionality and basic name changes. Upcoming PRs will convert the names of files, structs and functions to indicate this is now a generic tests.

**Release note**:
```

```

### CODE REVIEW

**Comment by kkmsft** [comment]
Note: Still getting some failures - 1 test is failing and one automated test is also failing. Working on figuring out what is going on.

File: test/e2e/suite_test.go [hunk 1]
```diff
@@ -35,13 +35,14 @@ import (
...
 	"github.com/pborman/uuid"
 	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/azurefile-csi-driver/test/e2e/driver"
 )
 
 const kubeconfigEnvVar = "KUBECONFIG"
 
 var azurefileDriver *azurefile.Driver
 
-func init() {
```
Comment (kkmsft) [inline test/e2e/suite_test.go]: With go 1.13.4, there are failure due to initializations of test flags. This change to BeforSuite solves the issue.

File: test/e2e/driver/azurefile_driver.go [hunk 2]
```diff
@@ -28,33 +30,52 @@ import (
...
-	return &azureFileCSIDriver{
-		driverName: azurefile.DriverName,
+// InitAzureFileDriver returns AzureFileDriver that implemnts DynamicPVTestDriver interface
+func InitAzureFileDriver() PVTestDriver {
+	driverName := os.Getenv(AzureDriverNameVar)
+	if driverName == "" {
+		driverName = azurefile.DriverName
+	}
+
+	klog.Infof("Using azure disk driver: %s", driverName)
```
Comment (andyzhangx) [inline test/e2e/driver/azurefile_driver.go]: file

File: test/e2e/README.md [hunk 3]
```diff
@@ -22,3 +22,11 @@ REGISTRY=<your Docker Hub ID / registry> make e2e-test
 # Run E2E tests on an existing Azure File CSI Driver image
 REGISTRY=<your Docker Hub ID / registry> IMAGE_VERSION=<desired image version> make e2e-test
 ```
+
+### Run the test for in-tree driver
+
+```
+export KUBECONFIG='<path to kubeconfig>'
+export AZURE_STORAGE_DRIVER="kubernetes.io/azure-disk"
```
Comment (andyzhangx) [inline test/e2e/README.md]: azure-file

File: test/e2e/suite_test.go [hunk 4]
```diff
@@ -50,81 +51,84 @@ func init() {
...
+		cmd := exec.Command("make", "e2e-bootstrap")
+		cmd.Dir = projectRoot
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		Expect(err).NotTo(HaveOccurred())
+		log.Println("Azure File CSI Driver installed")
+
+		nodeid := os.Getenv("nodeid")
+		Expect(nodeid).ShouldNot(BeEmpty())
```
Comment (chewong) [inline test/e2e/suite_test.go]: `nodeid` environment variable is not strictly required.

**Comment by andyzhangx** [comment]
@kkmsft 
 #125 and #128 are merged, could you rebase?

**Comment by kkmsft** [comment]
Reset and rebased on the latest.

Here is the latest run snapshot: 
Ran 5 of 8 Specs in 275.503 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 3 Skipped
--- PASS: TestE2E (275.50s)
PASS
ok

**Comment by andyzhangx** [comment]
@chewong  could you help enable `/test pull-kubernetes-e2e-azure-file` ? thanks.

**Pull request merged**